scripts/generate_dump

```diff
@@ -26,15 +35,96 @@ DO_COMPRESS=true
 CMD_PREFIX=
 SINCE_DATE="@0" # default is set to January 1, 1970 at 00:00:00 GMT
 REFERENCE_FILE=/tmp/reference
+TECHSUPPORT_TIME_INFO=/tmp/techsupport_time_info
 BASE=sonic_dump_`hostname`_`date +%Y%m%d_%H%M%S`
 DUMPDIR=/var/dump
 TARDIR=$DUMPDIR/$BASE
 TARFILE=$DUMPDIR/$BASE.tar
 LOGDIR=$DUMPDIR/$BASE/dump
 NUM_ASICS=1
+HOME=${HOME:-/root}
+USER=${USER:-root}
+TIMEOUT_MIN="5"
+SKIP_BCMCMD=0
+
+save_bcmcmd() {
+    local start_t=$(date +%s%3N)
+    local end_t=0
+    local cmd="$1"
+    local filename=$2
+    local filepath="${LOGDIR}/$filename"
+    local do_gzip=${3:-false}
+    local tarpath="${BASE}/dump/$filename"
+    local timeout_cmd="timeout --foreground ${TIMEOUT_MIN}m"
+    [ ! -d $LOGDIR ] && $MKDIR $V -p $LOGDIR
+
+    if [ $SKIP_BCMCMD -eq 1 ]; then
+        echo "Skip $cmd"
+        return 0
+    fi
+    # eval required here to re-evaluate the $cmd properly at runtime
+    # This is required if $cmd has quoted strings that should be bunched
+    # as one argument, e.g. vtysh -c "COMMAND HERE" needs to have
+    # "COMMAND HERE" bunched together as 1 arg to vtysh -c
+    if $NOOP; then
+        echo "${timeout_cmd} $cmd &> '${filepath}'"
+    else
+        eval "${timeout_cmd} $cmd" &> "${filepath}"
+        ret=$?
+        if [ $ret -ne 0 ]; then
+            if [ $ret -eq 124 ]; then
+                echo "Command: $cmd timedout after ${TIMEOUT_MIN} minutes."
+            else
+                grep "polling socket timeout: Success" ${filepath} &>/dev/null
+                if [ $? -eq 0 ]; then
+                    echo "bcmcmd command timeout. Setting SKIP_BCMCMD to true ..."
+                    SKIP_BCMCMD=1
+                fi
+            fi
+        fi
+    fi
+    if $do_gzip
+        gzip ${filepath} 2>/dev/null
+        tarpath="${tarpath}.gz"
+        filepath="${filepath}.gz"
+    fi
+    ($TAR $V -rhf $TARFILE -C $DUMPDIR "$tarpath" \
+        || abort "${ERROR_TAR_FAILED}" "tar append operation failed. Aborting to prevent data loss.") \
+        && $RM $V -rf "$filepath"
+    end_t=$(date +%s%3N)
+    echo "[ save_bcmcmd:$cmd ] : $(($end_t-$start_t)) msec" >> $TECHSUPPORT_TIME_INFO
+}
+
+###############################################################################
+# Runs a given bcmcmd command in all namesapces in case of multi ASIC platform
+# Globals:
+#  NUM_ASICS
+# Arguments:
+#  cmd: The command to run. Make sure that arguments with spaces have quotes
+#  filename: the filename to save the output as in $BASE/dump
+#  do_gzip: (OPTIONAL) true or false. Should the output be gzipped
+# Returns:
+#  None
+###############################################################################
+save_bcmcmd_all_ns() {
+    local do_gzip=${3:-false}
+
+    if [[ ( "$NUM_ASICS" > 1 ) ]]; then
+        for (( i=0; i<$NUM_ASICS; i++ ))
+        do
+            local cmd="bcmcmd -n $i $1"
```

Comment:
>bcmcmd -n [](start = 23, length = 9)

`bcmcmd` does not have `-n` option.

Reply:
The change was requested by @smaheshm.

bcmcmd has been modified to include the ASIC number. https://github.com/Azure/sonic-buildimage/pull/4926